### ISSUE TITLE
List Making

### ISSUE THREAD

**Issue by povrazor** [ludumdare/ludumdare#961]
This would be **SUPER USEFUL** to have. I have a few cases where I've needed to users to provide me list. If this was an actual feature, I could actually run queries on the raw lists I'm provided.

**Comment by local-minimum** [comment]
In those cases you talked about, you wanted the list to really be like:

```json
{
"title": "Games without content",
"description": "text"
"list":
  [
    {
      "title": "",
      "description": "",
      "checked": false,
    }
  ]
}
```

Where `checked` is optional to make checklist.
You'd easily want to add a site uri to a list of yours and automatically make new entry.

Besides useful for you, they would be a great way to curating games on the site.

I think also that there should be an auto-generating list for each event with the games you have voted on. Or at least a way to easily see that (like there used to be), but not disappear after voting is over.

**Comment by zwrawr** [comment]
#834 could probably share functionality with Polls/Surveys

**Comment by local-minimum** [comment]
Yes, it's just a matter of having some data of how it is published, I suppose.

**Comment by mikekasprzak** [comment]
Just as a general note, list-making needs to be very robust.

Most users will likely use it to manage a single global wishlist of things to check. This could see upwards of 1000 items for some of the most 

Some users will want to do separate lists per event.

Some users will want multiple lists per event. Streamers especially.

Sub-lists might even be something worth pursuing (i.e. a way to view everything on multiple lists at once). This almost makes them a feature similar to [Workflowy](https://workflowy.com). An alternative way of approaching this is to not have sub-lists, but rather let you define a user customizable `type` for each element of the list. Almost like tagging, though not necessarily tied to the existing (limited) tagging system.

List items should allow the list makers to include a note. A personalized blob of text.

Generally speaking, the listing feature should be approached similarly to Steam Curators.

http://store.steampowered.com/curator/537829-Ludum-Dare/

**Comment by mikekasprzak** [comment]
Lists should be a node, sharing the same pool as posts (i.e. belonging to a game or a user). This means they can have a description.

Should lists be an `item`? Maybe. `item/list`. This would mean they pollute a users item pool. They don't really need any of the benefits of items though, so maybe not.

Should lists be implemented using metadata (formerly links)? Maybe. A users wishlist could point to themselves, and take advantage of existing meta-link logic for figuring out how many lists I'm on and such. Yeah, this is probably a thing.

Using metadata means the following data can be written:
* `value` - arbitrary text
* `key` (slug) - the name of the metadata
* `a` and `b` - what node, and who (node) to associate it with

Things like sub-type filtering are not doable, as the name (key) is reserved by the field.

Also, is it worth polluting the metadata table further with additional data/indexes?

**Comment by mikekasprzak** [comment]
No, using metadata is not desirable. It means non-creators can pollute the raw data returned by the API. Lists should be isolated.

**Comment by local-minimum** [comment]
Our let's play community would really appreciate this, I think. And it shouldn't have to be too hard to implement something if we don't try too hard and make a perfect final solution in the first iteration.

I saw this tweet today https://twitter.com/s_standke/status/1123532562717188096 and thought, oh shit we really  need bookmarks. The way I see it lists are bookmarks+ so perhaps we could do a special list type that are "Games of event X" and have them show up on the `Event` -> `Me` section.

I put this comment here rather than on #1129 since this one seemed more detailed.
